### ISSUE TITLE
Add Support For Custom Optimizer and LR Scheduling in Trainer

### ISSUE THREAD

**Issue by mmilosevicTT** [tenstorrent/tt-blacksmith#643]
### Is your feature request related to a problem? Please describe

This feature is part of extending support of `Trainer` class.

Currently in all experiments we use `AdamW` optimizer without LR scheduling. This is pretty narrow and limits us to single optimizing technique. For `Trainer` class we need to move towards generality to enable different types of training that fit the user needs.

### Describe the solution you'd like

In PyTorchLightning we have `configure_optimizers` which gives us the option of setting custom optimizers. This approach is great, but lacks the support of configuring different optimizers without the need to know the python code. Main idea here is to make optimizer creation configurable through a config.

### Additional context

This PR should include:
- Addition of a config-like optimizer/lr-scheduler setting
- Addition of at least 2 different optimizer examples to confirm training is working
